index.d.ts

```diff
@@ -84,3 +84,26 @@ const ab: Merge<Foo, Bar> = {a: 1, b: 2};
 ```
 */
 export type Merge<FirstType, SecondType> = Omit<FirstType, Extract<keyof FirstType, keyof SecondType>> & SecondType;
+
+/*
+Create a type that requires at least one of the given keys.
+
+@example
+```
+import {RequireAtLeastOne} from 'type-fest';
+
+type SystemMessages = {
+	macos?: string;
+	linux?: string;
+	windows?: string;
+	default?: string;
+};
+
+const messages: RequireAtLeastOne<SystemMessages, 'macos' | 'linux' | 'windows'> = {macos: 'hey', default: 'hello'};
+```
+*/
+export type RequireAtLeastOne<T, Keys extends keyof T = keyof T> =
```

Comment:
Can you give `T` a better name? And `Keys` => `KeysType` (See the other types for naming inspiration).